crates/typst-cli/src/compile.rs

```diff
@@ -206,25 +241,26 @@ fn export_image(
     watching: bool,
     fmt: ImageExportFormat,
 ) -> StrResult<()> {
-    // Determine whether we have a `{n}` numbering.
     let output = command.output();
+    // Determine whether we have indexable templates in output
     let can_handle_multiple = match output {
         Output::Stdout => false,
-        Output::Path(ref output) => output.to_str().unwrap_or_default().contains("{n}"),
+        Output::Path(ref output) => {
+            OutputTemplate::has_indexable_template(output.to_str().unwrap_or_default())
+        }
     };
     if !can_handle_multiple && document.pages.len() > 1 {
-        let s = match output {
+        let path_err = format!(
+            "without a page number template ({}) in output path",
+            OutputTemplate::indexable_template().join(", ")
+        );
+        let err = match output {
             Output::Stdout => "to stdout",
-            Output::Path(_) => "without `{n}` in output path",
+            Output::Path(_) => path_err.as_str(),
```

Comment:
I think use `"to stdout".into()` and formatting in the match would be simpler. or `Cow` if we really care (I don't think we need to).